docs/site/repl-window.md

```diff
@@ -0,0 +1,151 @@
+
+## Find the File for the Current REPL Window Namespace
+
+When you are editing things in the REPL window, and want to open the file that defines its current namespace, use the **Show File for the Current REPL Window Namespace** command, `ctrl+alt+o r`.
+
+!!! Note
+    This also works for Clojure core and library namespaces.
+
+## Evaluating Code
+
+The window will be automatically associated with the REPL and the namespace of any project Clojure/ClojureScript file you evaluate code in. So for instance if you evaluate this code in a `clj` file with the namespace `fresh-reagent.handler`:
+
+```clojure
+(def mount-target
+  [:div#app
+   [:h2 "Welcome to fresh-reagent"]
+   [:p "please wait while Figwheel is waking up ..."]
+   [:p "(Check the js console for hints if nothing exciting happens.)"]])
+```
+
+If the REPL Window is set to be the output destination for evaluation results, the REPL window will print the defined var (otherwixe it will be printed to the destination configured). The REPL window will regardless print a new prompt reflecting the current REPL connection and namespace:
+
+![eval-results-1](images/howto/output/eval-results-1.png)
+
+If you then switch to the output window (`ctrl+alt+o r`), and enter this at the prompt:
+
+```clojure
+mount-target
+```
+
+then evaluate it using `alt+enter`, you'll get:
+
+![eval-results-2](images/howto/output/eval-results-2.png)
+
+This, since the namespace ”followed” the first evaluation over to the output window.
+
+## REPL History
+
+Recently evaluated forms in the REPL file are persisted and can easily be shown again for modifying and re-evaluating.
+
+### Navigate REPL History
+
+You can navigate up and down the last forms evaluated in the REPL file by using `alt+up` and `alt+down`, provided your cursor is at the end of the last form after the prompt. If the cursor is not at the end of the last form, then `alt+up` and `alt+down` will do what they are mapped to, which is by default "Move Line Up" and "Move Line Down," respectively.
+
+If you have typed some text after the prompt before you start traversing up the history, this text will be preserved and will display when you traverse back down the history. If you modify some text in the history while traversing, the modification will be saved at that location in history.
+
+### Clear REPL History
+
+You can clear the repl history by running the command "Clear REPL History" from the command palette.
+
+## Stack Traces
+
+When an evaluation produces an error, the output window will automatically print the the error message. If there is a stack trace associated with the error, this can now be printed on demand using the **Calva: Print Last Stacktrace to the Output Window** command. The output window will also have a Codelense button below the error message that will print the stack trace..
```

Comment:
```suggestion
When an evaluation produces an error, it will automatically be printed in the REPL Window. If there is a stack trace associated with the error, it can now be printed on demand using the **Calva: Print Last Stacktrace to the Output Window** command. You can also print the stack trace for any error message printed to the REPL Window via the Codelens button below it.
```